server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +278,104 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setRoutingWeights(
+        String awarenessAttributeName,
+        Map<String, Double> awarenessAttributeToWeightMap,
+        ActionListener<ClusterPutWeightedRoutingResponse> listener
+    ) {
+        // WRR API will validate invalid weights
+        final ClusterPutWeightedRoutingRequest clusterPutRoutingWeightRequest = new ClusterPutWeightedRoutingRequest(
+            awarenessAttributeName
+        );
+        clusterPutRoutingWeightRequest.setWeightedRouting(new WeightedRouting(awarenessAttributeName, awarenessAttributeToWeightMap));
+
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            ClusterAddWeightedRoutingAction.NAME,
+            clusterPutRoutingWeightRequest,
+            new TransportResponseHandler<ClusterPutWeightedRoutingResponse>() {
+                @Override
+                public void handleResponse(ClusterPutWeightedRoutingResponse response) {
+                    logger.info("Weights are successfully set. [{}]", clusterPutRoutingWeightRequest.getWeightedRouting().weights());
```

Comment:
Do we need to use toString?

Reply:
Not needed. It will be called implicitly

Reply:
Can we directly use `WeightedRoutingService#registerWeightedRoutingMetadata` instead